io_scene_godot/structures.py

```diff
@@ -132,21 +134,22 @@ def generate_heading_string(self):
     def generate_body_string(self):
         """Convert the contents of the super/internal dict into newline
         separated key=val pairs"""
-        out_str = ''
+        lines = []
         for var in self:
             val = self[var]
             val = to_string(val)
-
-            out_str += '\n{} = {}'.format(var, val)
-        return out_str
+            lines.append('{} = {}'.format(var, val))
+        return "\n".join(lines)
 
     def to_string(self):
         """Serialize this entire entry"""
-        return "{}\n{}{}".format(
-            self.generate_heading_string(),
-            self.generate_body_string(),
-            self.contents
-        )
+        heading = self.generate_heading_string()
```

Comment:
I understand your changing would make the generated file more formatted. However I kind of believe an extra empty line does not matter, I would be much happier to make the code more simple and straightforward. So how about keep the original one?

Reply:
Or maybe
```
        heading = self.generate_heading_string()
        body = self.generate_body_string()
        if body and self.contents:
            return "{}\n\n{}\n{}".format(heading, body, self.contents)
        if body:
            return "{}\n\n{}".format(heading, body)
        return heading
```